examples/js/controls/VRControls.js

```diff
@@ -9,22 +9,28 @@ THREE.VRControls = function ( object, onError ) {
 
 	var vrInputs = [];
 
-	function filterInvalidDevices( devices ) {
+function filterInvalidDevices( devices ) {
 
 		// Exclude Cardboard position sensor if Oculus exists.
 
 		var oculusDevices = devices.filter( function ( device ) {
-
-			return device.deviceName.toLowerCase().indexOf( 'oculus' ) !== - 1;
+			if(device.deviceName) {
+				return device.deviceName.toLowerCase().indexOf( 'oculus' ) !== - 1;
+			} else {
+				return [];
```

Comment:
In that case, the function filterInvalidDevices should be changes so it
doesn't assume the returned value has a .length. That would also fix the
broken promise.

On Tue, Dec 29, 2015 at 7:23 AM, Mr.doob <notifications@github.com> wrote:

> In examples/js/controls/VRControls.js
> <https://github.com/mrdoob/three.js/pull/7862#discussion_r48546676>:
>
> >
> >  		// Exclude Cardboard position sensor if Oculus exists.
> >
> >  		var oculusDevices = devices.filter( function ( device ) {
> > -
> > -			return device.deviceName.toLowerCase().indexOf( 'oculus' ) !== - 1;
> > +			if(device.deviceName) {
> > +				return device.deviceName.toLowerCase().indexOf( 'oculus' ) !== - 1;
> > +			} else {
> > +				return [];
>
> I think the function passed to Array.filter() is supposed to return a
> boolean.
>
> https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mrdoob/three.js/pull/7862/files#r48546676>.
>



-- 
============================================================
=================
*Dr. Pete Markiewicz*
*Email:* pindiespace@gmail.com
*Sustainable virtual design blog:*
http://sustainablevirtualdesign.wordpress.com
*Sustainability template:* http://greenboilerplate.com
*Teaching site: *http://plyojump.com <http://www.plyojump.com/>
*Buy my book! *- Millennials and the Popular Culture
*Lifecourse:* http://www.lifecourse.com/store/catalog/lca/mpc.html
* AmazonAuthor Page:*
http://www.amazon.com/Pete-Markiewicz/e/B0069VZ7BG/ref=ntt_athr_dp_pel_pop_2
*On that page, see my DARPA Grand Challenge 2005 entry video, "Team
Robomonster":*
*Team Robomonster *(DARPA 2005 Grand Challenge):
http://robotsthatjump.wordpress.com/category/team-robomonster/
*Indiespace *(1st to sell downloaded music, March 1994):
http://www.indiespace.com. Also see http://archive.org under
"Kspace.com" 1996 for our original site.
============================================================
=================